libraries/AP_Terrain/AP_Terrain.cpp

```diff
@@ -309,12 +304,9 @@ void AP_Terrain::update(void)
     // just schedule any needed disk IO
     schedule_disk_io();
 
-    // try to ensure the home location is populated
-    float height;
-    height_amsl(ahrs.get_home(), height, false);
```

Comment:
I've actually dismissed this as intentional in Coverity two weeks ago. This is what I wrote there:

> height_amsl has a special handling for home location, so this tries to set the terrain height at home whenever is possible - for example, terrain height might not available when home is set but be available at a later time, this would guarantee we will have it as fast as possible.